client/HubApi.ts

```diff
@@ -84,28 +84,28 @@ export default class HubApi<DB extends BehaviorType = BehaviorType.POPUP> { // D
      */
 
     public checkout<B extends BehaviorType = DB>(
-        request: CheckoutRequest,
+        request: Promise<CheckoutRequest>,
```

Comment:
When the type is only promise now, is it still OK and not a type error when a regular object is passed?

Reply:
yes

Reply:
Nope, that's not true. `CheckoutRequest` objects can't be passed instead of `Promise<CheckoutRequest>` as `CheckoutRequest` doesn't have `then` or `catch`.